images/router/clear-route-status.sh

```diff
@@ -0,0 +1,117 @@
+    MODIFIED_JSON=$(echo "$MY_JSON_BLOB" | jq 'del(.status.ingress)')
+    curl -s -X PUT http://localhost:8001/oapi/v1/namespaces/${namespace}/routes/${route_name}/status --data-binary "${MODIFIED_JSON}" -H "Content-Type: application/json" > /dev/null
+    echo "route status for route ${route_name} in namespace ${namespace} cleared"
+}
+
+#sets up clearing a status set by a specific router
+function clear_status_set_by() {
+    local router_name="$1"
+
+    NAMESPACES=$(oc get namespaces -o jsonpath='{range .items[*]}{.metadata.name} {end}')
+    for NAMESPACE in $NAMESPACES; do
+        ROUTES=$(oc get routes -o jsonpath='{range .items[*]}{.metadata.name} {end}' --namespace=$NAMESPACE 2>/dev/null)
+        if [[ -n "${ROUTES}" ]]; then
+            for ROUTE in $ROUTES; do
+                clear_routers_status $NAMESPACE $ROUTE $router_name
+            done
+        else
+            echo "No routes found for namespace ${NAMESPACE}"
+        fi
+    done
+
+}
+
+# clears the status field of a specific router name
+function clear_routers_status() {
+    local namespace="$1"
+    local route_name="$2"
+    local router_name="$3"
+    MY_JSON_BLOB=$(oc get --raw http://localhost:8001/oapi/v1/namespaces/${namespace}/routes/${route_name}/) 
+    INDEX=$(echo "$MY_JSON_BLOB" | jq '.status.ingress | map(.routerName != "'${router_name}'") | index(false)')
+    if [[ "${INDEX}" != null ]]; then
+        MODIFIED_JSON=$(echo "$MY_JSON_BLOB" | jq 'del(.status.ingress['${INDEX}'])')
+        curl -s -X PUT http://localhost:8001/oapi/v1/namespaces/${namespace}/routes/${route_name}/status --data-binary "${MODIFIED_JSON}" -H "Content-Type: application/json" > /dev/null
+        echo "route status for route ${route_name} set by router ${router_name} cleared"
+    else
+        echo "route ${route_name} has no status set by ${router_name}"
+    fi
+}
+
+cleanup() {
+    if [[ -n "${PROXY_PID:+unset_check}" ]]; then
+        kill $PROXY_PID
+    fi
+}
+trap cleanup EXIT
+
+USAGE="Usage:
+To clear only the status set by a specific router on all routes in all namespaces
+./clear-router-status.sh -r [router_name]
+
+router_name is the name in the deployment config, not the name of the pod. If the router is running it will
+immediatly update any cleared status.
+
+To clear the whole status field of the route by route name 
+./clear-route-status.sh [namespace] [route-name | ALL]
+
+To clear the status of all routes in all namespaces:
+oc get namespaces | awk '{if (NR!=1) print \$1}' | xargs -n 1 -I %% ./clear-route-status.sh %% ALL
+
+To clear the status of all routes in namespace default:
+./clear-route-status.sh defalt ALL
+
+To clear the status of route example in namespace default:
+./clear-route-status.sh default example
+
+NOTE: if a router that admits a route is running it will immediatly update the cleared route status 
+"
+
+if [[ ${#} -ne 2 || "${@}" == *" help "* ]]; then
+    printf "$USAGE"
```

Comment:
You're using `${USAGE}` as the format string for `printf` here ... either:

```sh
pritnf "%s" "${USAGE}"
```

or

```sh
echo -n "${USAGE}"
```